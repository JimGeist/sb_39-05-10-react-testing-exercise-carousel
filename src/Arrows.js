import React from "react";
import "./Arrows.css";

function ArrowBackwards({ fx = "", disable = "" }) {

    if (typeof (fx) === "function") {
        // return enabled left arrow component with onClick function
        return (
            <i
                className={`fas fa-chevron-circle-left fa-2x`}
                onClick={fx}
                data-testid="left-arrow"
            />
        );
    } else {
        // return disabled left arrow component without onClick function
        return (
            <i
                className={`fas fa-chevron-circle-left fa-2x ${disable}`}
                data-testid="left-arrow"
            />
        );
    }

}

function ArrowForwards({ fx = "", disable = "" }) {

    if (typeof (fx) === "function") {
        // return enabled right arrow component with onClick function
        return (
            <i
                className={`fas fa-chevron-circle-right fa-2x`}
                onClick={fx}
                data-testid="right-arrow"
            />
        );
    } else {
        // return disabled right arrow component without onClick function
        return (
            <i
                className={`fas fa-chevron-circle-right fa-2x ${disable}`}
                data-testid="right-arrow"
            />
        );
    }


}

export { ArrowBackwards, ArrowForwards };
