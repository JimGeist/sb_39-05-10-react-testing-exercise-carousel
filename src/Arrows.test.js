import React from "react";
import { render } from "@testing-library/react";
import { ArrowBackwards, ArrowForwards } from "./Arrows";

const goBackward = () => 1 - 1;
const goForward = () => 1 + 1;

// Smoke Test -- does ArrowBackwards without props render?
it("renders the backwards arrow without crashing", () => {
    render(<ArrowBackwards />);
});

// Smoke Test -- does ArrowBackwards with fx render?
it("renders the backwards arrow with onClick fx without crashing", () => {
    render(<ArrowBackwards fx={goBackward} />);
});

// ArrowBackwards disabled snapshot test
it("Disabled ArrowBackwards matches snapshot", () => {
    const { asFragment } = render(<ArrowBackwards fx="" disable="Arrows-disabled" />);
    expect(asFragment()).toMatchSnapshot();
});

it("Enabled ArrowBackwards with onClick fx matches snapshot", () => {
    const { asFragment } = render(<ArrowBackwards fx={goBackward} disable="" />);
    expect(asFragment()).toMatchSnapshot();
});

// Smoke Test -- does ArrowForwards without props render?
it("renders the forwards arrow without crashing", () => {
    render(<ArrowForwards />);
});

// Smoke Test -- does ArrowBackwards with fx render?
it("renders the forwards arrow with onClick fx without crashing", () => {
    render(<ArrowForwards fx={goForward} />);
});

// ArrowForwards disabled snapshot test
it("Disabled ArrowForwards matches snapshot", () => {
    const { asFragment } = render(<ArrowForwards fx="" disable="Arrows-disabled" />);
    expect(asFragment()).toMatchSnapshot();
});

it("Enabled ArrowForwards with onClick fx matches snapshot", () => {
    const { asFragment } = render(<ArrowForwards fx={goForward} disable="" />);
    expect(asFragment()).toMatchSnapshot();
});