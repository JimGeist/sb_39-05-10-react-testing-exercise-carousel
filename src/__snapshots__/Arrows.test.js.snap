// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabled ArrowBackwards matches snapshot 1`] = `
<DocumentFragment>
  <i
    class="fas fa-chevron-circle-left fa-2x Arrows-disabled"
    data-testid="left-arrow"
  />
</DocumentFragment>
`;

exports[`Disabled ArrowForwards matches snapshot 1`] = `
<DocumentFragment>
  <i
    class="fas fa-chevron-circle-right fa-2x Arrows-disabled"
    data-testid="right-arrow"
  />
</DocumentFragment>
`;

exports[`Enabled ArrowBackwards with onClick fx matches snapshot 1`] = `
<DocumentFragment>
  <i
    class="fas fa-chevron-circle-left fa-2x"
    data-testid="left-arrow"
  />
</DocumentFragment>
`;

exports[`Enabled ArrowForwards with onClick fx matches snapshot 1`] = `
<DocumentFragment>
  <i
    class="fas fa-chevron-circle-right fa-2x"
    data-testid="right-arrow"
  />
</DocumentFragment>
`;
